### ISSUE TITLE
prefer-ternary: option to ignore when multiple lines

### ISSUE THREAD

**Issue by noftaly** [sindresorhus/eslint-plugin-unicorn#889]
There should be an option for the rule prefer-ternary to ignore when there are multiple lines.
I just updated my config, and it reported me that this was wrong:

```js
if (process.platform === 'win32') {
    await exec(oneLine`
        A long line,
        which is on multiple lines
        etc etc
    `, { cwd: paths.project });
} else {
     await exec(oneLine`
          Same, but
          not on windows,
          etcetc
      `, { cwd: paths.project });
}
```

I think ternary expressions are really great for short lines, like in the examples on the docs, but for long lines like this, it should be skipped.

**Comment by fisker** [comment]
It fixes to

```js
await (
	process.platform === 'win32') ?
    exec(oneLine`
        A long line,
        which is on multiple lines
        etc etc
    `, { cwd: paths.project }) :
     exec(oneLine`
          Same, but
          not on windows,
          etcetc
      `, { cwd: paths.project });
)
```

right?

I don't think this is still readable.

I would write this case to

```js
const command = process.platform === 'win32' ? 
	oneLine`
        A long line,
        which is on multiple lines
        etc etc
    ` :
	oneLine`
        Same, but
        not on windows,
        etcetc
    `;

await exec(command, { cwd: paths.project })
```

Ideally, we should merge all common parts, this is what I want to do, but it's hard.


Of course, there are some other cases too complicated to merge.

I'm open to the option.

**Comment by noftaly** [comment]
It fixed to
```js
await (process.platform === 'win32' ? exec(oneLine`
          A long line,
          which is on multiple lines
          etc etc
        `, { cwd: paths.project }) : exec(oneLine`
          Same, but
          not on windows,
          etcetc
        `, { cwd: paths.project }));
```
which I find way less readable. It's true that I can then refactor it a bit to match your first code.

I did not think of the second option and it true that it is way more readable! I changed it to that. I guess the option makes less sense then... Maybe still for long method chaining?

**Comment by szmarczak** [comment]
@sindresorhus Friendly ping :) Another reproducible case:

```js
			if (alpnProtocol === 'h2') {
				agent = {
					http: new HttpOverHttp2(proxy),
					https: new HttpsOverHttp2(proxy),
					http2: new Http2OverHttp2(proxy)
				};
			} else {
				agent = {
					http: new HttpsProxyAgent(proxy.url.href),
					https: new HttpsProxyAgent(proxy.url.href),
					http2: new Http2OverHttps(proxy)
				};
			}
```

**Comment by sindresorhus** [comment]
Any suggestions on what the option should be called?

**Comment by szmarczak** [comment]
`single-line` / `multiple-lines`

**Comment by sindresorhus** [comment]
I think it would be more consistent with built-in ESLint rules to use: `always` / `only-single-line` (Like https://eslint.org/docs/rules/comma-dangle)

**Comment by szmarczak** [comment]
Agreed
